### ISSUE TITLE
Array size off by one in morris optimal trajectories

### ISSUE THREAD

**Issue by jdherman** [jdherman/SALib#27]
From @saroele ...

```
And then I have some bad news: I get a bug when trying ``optimal_trajectories=4``. 
With 2, 3 and 5 it works (the latter takes a LOT of time though, but that is documented).  

# Generate samples 
param_values = Morris(os.path.join('results','parameters_'+s_id+'.txt'), 
                  samples=100,
                  num_levels=num_levels, 
                  grid_jump=grid_jump,
                  optimal_trajectories=4)

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-22-be5a7e2544a6> in <module>()
      4                       num_levels=num_levels,
      5                       grid_jump=grid_jump,
----> 6                       optimal_trajectories=4)
      7 #print param_values
      8 

/home/roel/data/work/SALib/SALib/sample/morris.pyc in __init__(self, parameter_file, samples, num_levels, grid_jump, group_file, optimal_trajectories)
     62         if self.groups is None:
     63 
---> 64             self.create_sample()
     65 
     66         else:

/home/roel/data/work/SALib/SALib/sample/morris.pyc in create_sample(self)
    129                                                          self.samples,
    130                                                          self.num_vars,
--> 131                                                          self.optimal_trajectories)
    132 
    133         self.output_sample = optimal_sample

/home/roel/data/work/SALib/SALib/sample/morris_optimal.pyc in find_optimum_trajectories(input_sample, N, num_params, k_choices)
    132                                N,
    133                                num_params,
--> 134                                k_choices)
    135 
    136     index_list = []

/home/roel/data/work/SALib/SALib/sample/morris_optimal.pyc in find_most_distant(input_sample, N, num_params, k_choices)
     72 
     73     for combos in grouper(chunk, combo_gen):
---> 74         scores[(counter*chunk):((counter+1)*chunk)] = mappable(combos, pairwise, distance_matrix)
     75         counter += 1
     76     return scores

ValueError: could not broadcast input array from shape (921225) into shape (921224)
```

Another example ...

```
Similar bug with a different combination : 
 # Generate samples 
 param_values = Morris(os.path.join('results','parameters_'+s_id+'.txt'), 
                  samples=200,
                  num_levels=num_levels, 
                  grid_jump=grid_jump,
                  optimal_trajectories=3)

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-32-40d2028baf35> in <module>()
      4                       num_levels=num_levels,
      5                       grid_jump=grid_jump,
----> 6                       optimal_trajectories=3)
      7 #print param_values
      8 

/home/roel/data/work/SALib/SALib/sample/morris.pyc in __init__(self, parameter_file, samples, num_levels, grid_jump, group_file, optimal_trajectories)
     62         if self.groups is None:
     63 
---> 64             self.create_sample()
     65 
     66         else:

/home/roel/data/work/SALib/SALib/sample/morris.pyc in create_sample(self)
    129                                                          self.samples,
    130                                                          self.num_vars,
--> 131                                                          self.optimal_trajectories)
    132 
    133         self.output_sample = optimal_sample

/home/roel/data/work/SALib/SALib/sample/morris_optimal.pyc in find_optimum_trajectories(input_sample, N, num_params, k_choices)
    132                                N,
    133                                num_params,
--> 134                                k_choices)
    135 
    136     index_list = []

/home/roel/data/work/SALib/SALib/sample/morris_optimal.pyc in find_most_distant(input_sample, N, num_params, k_choices)
     72 
     73     for combos in grouper(chunk, combo_gen):
---> 74         scores[(counter*chunk):((counter+1)*chunk)] = mappable(combos, pairwise, distance_matrix)
     75         counter += 1
     76     return scores

ValueError: could not broadcast input array from shape (313400) into shape (313399)
```

**Comment by jdherman** [comment]
Hi @saroele -- happy new year! Did this array size issue continue to be a problem for you? If so, could you share your parameter file and the inputs you're using for this function call?
```python
param_values = Morris(os.path.join('results','parameters_'+s_id+'.txt'), 
                  samples=100,
                  num_levels=num_levels, 
                  grid_jump=grid_jump,
                  optimal_trajectories=4)
```
I'll try to reproduce it. As an aside, the Morris API is back to a single `sample()` function rather than a class. [Here's an example](https://github.com/jdherman/SALib/blob/master/examples/morris/morris.py). Shouldn't have any effect on this bug, though.

**Comment by saroele** [comment]
Hi Jon,

I don't like saying it, but I won' thave time the coming weeks to dig into
it.  I'm sorry.  We have been able to use SALib and are very grateful for
your work.
Maybe I can come back to this in april, but untill then don't count on me.

roel

**Comment by jdherman** [comment]
@saroele, no problem!! I just wanted to make sure this bug wasn't preventing you from doing your work. I haven't seen it reported elsewhere, so if it's working for you then I won't try to track it down now.

**Comment by willu47** [comment]
I landed this same bug again using numpy 1.7.1 on win-32 with python 2.7.

**Comment by jdherman** [comment]
As of #40 we've bumped the numpy dependency to 1.8 or later. This doesn't fix the bug, but seems good enough to close the issue for now.
